### PR TITLE
[API][ChannelPriceHistoryConfig] Upgrade and embed into channel

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A
|-----------------|-----
| Branch?         | api-platform-3
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ChannelPriceHistoryConfigDenormalizer.php [hunk 1]
```diff
@@ -11,29 +11,26 @@
...
-use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
 use Symfony\Component\Serializer\Normalizer\DenormalizerAwareInterface;
 use Symfony\Component\Serializer\Normalizer\DenormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
 use Webmozart\Assert\Assert;
 
-final class ChannelPriceHistoryConfigDenormalizer implements ContextAwareDenormalizerInterface, DenormalizerAwareInterface
+final class ChannelPriceHistoryConfigDenormalizer implements DenormalizerInterface, DenormalizerAwareInterface
 {
     use DenormalizerAwareTrait;
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Serializer/Denormalizer/ChannelPriceHistoryConfigDenormalizer.php:27]:
```suggestion
final class ChannelPriceHistoryConfigDenormalizer implements DenormalizerInterface
{
```

File: src/Sylius/Behat/Resources/config/services/contexts/api/admin.xml [hunk 2]
```diff
@@ -316,5 +316,12 @@
 <!--            <argument type="service" id="Sylius\Behat\Client\ResponseCheckerInterface" />-->
 <!--            <argument type="service" id="sylius.behat.clock" />-->
 <!--        </service>-->
+
+        <service id="Sylius\Behat\Context\Api\Admin\ManagingChannelPriceHistoryConfigContext">
```
Comment (Wojdylak) [inline src/Sylius/Behat/Resources/config/services/contexts/api/admin.xml:320]: Dotted notation

File: src/Sylius/Behat/Context/Api/Admin/ManagingChannelPriceHistoryConfigContext.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Service\Converter\IriConverterInterface;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Webmozart\Assert\Assert;
+
+final class ManagingChannelPriceHistoryConfigContext implements Context
```
Comment (Wojdylak) [inline src/Sylius/Behat/Context/Api/Admin/ManagingChannelPriceHistoryConfigContext.php:26]:
```suggestion
final readonly class ManagingChannelPriceHistoryConfigContext implements Context
```

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Admin/ManagingChannelPriceHistoryConfigContext.php [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    /**
+     * @Then /^the "[^"]+" channel should have the lowest price of discounted products prior to the current discount (enabled|disabled)$/
+     */
+    public function theChannelShouldHaveTheLowestPriceOfDiscountedProductsPriorToTheCurrentDiscountEnabledOrDisabled(
+        string $visible,
+    ): void {
+        Assert::same(
+            $this->getChannelPricingFieldFromLastResponse('lowestPriceForDiscountedProductsVisible'),
+            $visible === 'enabled',
+        );
```
Comment (TheMilek) [inline src/Sylius/Behat/Context/Api/Admin/ManagingChannelPriceHistoryConfigContext.php:109]:
```suggestion
        Assert::same(
            $this->getChannelPricingFieldFromLastResponse('lowestPriceForDiscountedProductsVisible'),
            $visible === 'enabled',
            sprintf('This channel should have the lowest price of discounted products prior to the current discount %s, but have not', $visible)
        );
```
lets add this error messages everywhere

**Review by TheMilek — APPROVED** [review]

**Review by mpysiak — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Admin/ManagingChannelPriceHistoryConfigContext.php [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    }
+
+    /**
+     * @Then /^the "[^"]+" channel should have the lowest price for discounted products checking period set to (\d+) days$/
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingChannelPriceHistoryConfigContext.php:113]: Is this step worked correctly as the channel name is not defined as a method argument? 🤔

**Review by GSadee — APPROVED** [review]

**Pull request merged**